torch_xla/csrc/helpers.cpp

```diff
@@ -54,9 +56,10 @@ xla::int64 XlaHelpers::GetCanonicalDimensionIndex(xla::int64 dim,
                                                   xla::int64 rank) {
   xla::int64 min_shape_dim = -rank;
   xla::int64 max_shape_dim = rank - 1;
-  XLA_CHECK(min_shape_dim <= dim && dim <= max_shape_dim)
-      << "Value out of range (expected to be in range of [" << min_shape_dim
-      << ", " << max_shape_dim << "], but got " << dim << ")";
+  if (min_shape_dim > dim || dim > max_shape_dim) {
+    AT_INDEX_ERROR("Value out of range (expected to be in range of [",
```

Comment:
Argh :smile: , so not only matching strings, but also exception types?
I love enabling tests for which we generate wrong results, but the ones which fails only because of a regex match or exception type ... not sure.

Reply:
We had a discussion a long time ago whether we should completely replicate PyTorch exception types and messages and we decided to unify error handling and use our own. Maybe the test can be more lenient here?